### PR TITLE
Flatten Maven project

### PR DESCRIPTION
**Pull request by basil**
### Problem

This component originally had a number of Maven modules: a model which was then consumed by a CLI and a Google App Engine client. The Google App Engine client was ripped out at some point, so now the multi-module build remains, even though in practice there is really just one artifact that is produced by this repository, which is the PCT command-line interface (CLI). I could not find any consumers of any artifacts other than the CLI after searching usages in sources and binaries in both the `jenkinsci` and `cloudbees` GitHub organizations. As such, the split into multiple modules just makes it more difficult to edit the code and update dependencies.

### Solution

Flatten the project into a single CLI module.

### Implementation

I essentially just merged the contents of the POM files together. There is no new code here, just moved code.

### Testing done

Testing in `jenkinsci/bom` complete at https://github.com/jenkinsci/bom/pull/1678. I am hoping that James can run the same test for any proprietary consumers.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +54,12 @@
         <version>1.6</version>
       </dependency>
 
+      <dependency>
+        <groupId>com.google.code.findbugs</groupId>
+        <artifactId>annotations</artifactId>
+        <version>3.0.1u2</version>
+      </dependency>
```
Comment (basil) [inline pom.xml:61]: Moved from `plugins-compat-tester-model/pom.xml`.

File: pom.xml [hunk 2]
```diff
@@ -82,6 +84,12 @@
         <version>1.7.25</version>
       </dependency>
 
+      <dependency>
+        <groupId>net.java.dev.jna</groupId>
+        <artifactId>jna</artifactId>
+        <version>4.1.0</version>
+      </dependency>
```
Comment (basil) [inline pom.xml:91]: Added to resolve an upper bounds conflict.

File: pom.xml [hunk 3]
```diff
@@ -136,4 +312,59 @@
...
+                    <exclude>META-INF/*.RSA</exclude>
+                  </excludes>
+                </filter>
+              </filters>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
```
Comment (basil) [inline pom.xml:369]: Moved from `plugins-compat-tester-cli/pom.xml`.

File: pom.xml [hunk 4]
```diff
@@ -115,11 +123,179 @@
   </dependencyManagement>
 
   <dependencies>
+    <dependency>
```
Comment (basil) [inline pom.xml:126]: Everything added here was moved from one of the other `pom.xml` files.

File: plugins-compat-tester/pom.xml [hunk 5]
```diff
@@ -1,198 +0,0 @@
...
-      <version>1.5</version>
-  	</dependency>
-	<dependency>
-	  <groupId>org.apache.maven</groupId>
-	  <artifactId>maven-model</artifactId>
-	  <version>3.0.2</version>
-	</dependency>
-  	
-
-    <!--TODO(oleg_nenashev): Remove all providers except Git? -->
```
Comment (jglick) [inline plugins-compat-tester/pom.xml:68]: Note: merge conflict with #382.

File: pom.xml [hunk 6]
```diff
@@ -28,22 +28,18 @@
     <gitHubRepo>jenkinsci/plugin-compat-tester</gitHubRepo>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <logbackVersion>1.2.3</logbackVersion>
+    <maven.scm.providers.version>1.5</maven.scm.providers.version>
+    <trimStackTrace>false</trimStackTrace> <!-- SUREFIRE-1798 -->
```
[the remark this answers is not in the document]
Comment (jglick) [reply pom.xml:32]: BTW I think this is either inherited from the parent POM, or (if the Surefire version is inherited) obsolete.

**Review by jglick — APPROVED** [review]
Thanks! This does seem like a long-overdue simplification.

File: pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (basil) [reply pom.xml:32]: BTW you never found the time to follow up on https://github.com/jenkinsci/plugin-compat-tester/pull/348#issuecomment-1059283158.

[the remark this answers is not in the document]
Comment (basil) [reply pom.xml:32]: Is `trimStackTrace` somehow related to flattening the Maven project? :man_shrugging:

[the remark this answers is not in the document]
Comment (jglick) [reply pom.xml:32]: Just to the extent that it is POM text you are copying, and you might not have been aware that it is (IIRC) obsolete and could probably be omitted from the copy. If you did know and do not care, then you can simply ignore.

[the remark this answers is not in the document]
Comment (basil) [reply pom.xml:32]: A workaround you introduced in https://github.com/jenkinsci/plugin-pom/pull/33 and which propagated to other components, and which you never removed once it was no longer needed? Not of concern to me. Do not bother me with such things, especially if you are not willing to follow up on https://github.com/jenkinsci/plugin-compat-tester/pull/348#issuecomment-1059283158 and similar.

**Review by jtnord — APPROVED** [review]

**Comment by jtnord** [comment]
local code adapted and seems to be progressing well.

**Comment by basil** [comment]
> local code adapted and seems to be progressing well.

Great!

> I think you have permission to merge? If not, I will.

I do not require your assistance.

**Pull request merged**